Fix diff parsing to support mnemonicPrefix configuration

MirrorDNA-Reflection-Protocol opened the pull request:
## Summary

Fixes #2013

When `diff.mnemonicPrefix=true` is set in git config, git uses different prefixes for diff paths instead of the standard `a/` and `b/`:
- `c/` for commit
- `w/` for worktree
- `i/` for index
- `o/` for object
- `h/` for HEAD

Previously, the diff regex and `decode_path()` function only accepted `a/` and `b/` prefixes, causing `create_patch=True` diffs to fail parsing when users had this config enabled.

## Reproduction

As described in #2013:
```python
# With ~/.gitconfig containing:
# [diff]
#     mnemonicPrefix = true

repo = git.Repo('.')
ancestor_ref = repo.merge_base(repo.head, repo.refs['main'])
diff = ancestor_ref[0].diff(None, create_patch=True)  # Would fail to parse
```

## Changes

- Update `re_header` regex to accept `[abciwoh]/` prefixes instead of just `[ab]/`
- Update `decode_path()` assertion to accept all valid mnemonic prefixes
- Add test case for mnemonicPrefix-style diffs

## Testing

- New test `test_diff_mnemonic_prefix` verifies parsing works with `c/` and `w/` prefixes
- All existing diff tests pass (except one pre-existing failure on main unrelated to this change)

## References

- Git documentation: https://git-scm.com/docs/git-config#Documentation/git-config.txt-diffmnemonicPrefix
- Issue: #2013

MirrorDNA-Reflection-Protocol committed:
Fix diff parsing to support mnemonicPrefix configuration

Fixes #2013

When diff.mnemonicPrefix=true is set in git config, git uses different
prefixes for diff paths instead of the standard a/ and b/:
- c/ for commit
- w/ for worktree
- i/ for index
- o/ for object
- h/ for HEAD

Previously, the diff regex and decode_path() function only accepted
a/ and b/ prefixes, causing create_patch=True diffs to fail parsing.

Changes:
- Update re_header regex to accept [abciwoh]/ prefixes
- Update decode_path() assertion to accept all valid mnemonic prefixes
- Add test case for mnemonicPrefix-style diffs

MirrorDNA-Reflection-Protocol committed:
Fix lint and mypy errors

- Remove whitespace from blank lines in docstrings (ruff W293)
- Use !r format specifier for bytes in f-string (mypy str-bytes-safe)

Byron reviewed (changes requested):
Thanks a lot for contributing! I particularly like the exhaustive documentation around the matter, and what I see looks good!

The test is also something I'd love to see improved, and I side with Copilot here.

MirrorDNA-Reflection-Protocol committed:
Address review feedback: improve test coverage for all mnemonicPrefix values

Changes per reviewer (Byron) and Copilot suggestions:
- Add h/ (HEAD) to the comment listing supported prefixes
- Update test docstring to include h/ prefix
- Expand test to cover all prefix combinations using subTest:
  - c/ (commit) vs w/ (worktree)
  - c/ (commit) vs i/ (index)
  - i/ (index) vs w/ (worktree)
  - o/ (object) vs w/ (worktree)
  - h/ (HEAD) vs i/ (index)
  - h/ (HEAD) vs w/ (worktree)

This ensures the regex pattern [abciwoh] and decode_path() work correctly
with all supported mnemonicPrefix values.
